src/platform/with_wgl/native_gl_context.rs

```diff
@@ -0,0 +1,144 @@
+use platform::NativeGLContextMethods;
+use std::ffi::CString;
+use std::os::raw::{c_void};
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+use winapi;
+use user32;
+use kernel32;
+use super::wgl;
+use super::wgl_attributes::*;
+use gleam::gl;
+
+/// Wrapper to satisfy `Sync`.
+struct HMODULEWrapper(winapi::HMODULE);
+unsafe impl Sync for HMODULEWrapper {}
+
+lazy_static! {
+    static ref GL_LIB: Option<HMODULEWrapper>  = {
+        let p = unsafe{kernel32::LoadLibraryA(b"opengl32.dll\0".as_ptr() as *const _)};
+        if p.is_null() {
+            debug!("opengl32.dll not found!");
+            None
+        }
+        else {
+            debug!("opengl32.dll LOADED!");
+            Some(HMODULEWrapper(p))
+        }
+    };
+}
+
+static LOAD_GL: Once = ONCE_INIT;
+pub fn load_gl() {
+    LOAD_GL.call_once(|| {
+        gl::load_with(|s| {
+            NativeGLContext::get_proc_address(s) as *const _
+        });
+    });
+}
+
+pub struct NativeGLContext {
+    pub render_ctx: winapi::HGLRC,
+    pub device_ctx:winapi::HDC, //
+    pub weak: bool,
+}
+
+impl Drop for NativeGLContext {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.weak {
+                wgl::DeleteContext(self.render_ctx as *const _);
+                let window = user32::WindowFromDC(self.device_ctx);
+                user32::ReleaseDC(window, self.device_ctx);
+                user32::DestroyWindow(window);
+            }
+        }
+    }
+}
+
+pub struct NativeGLContextHandle(pub winapi::HGLRC, pub winapi::HDC);
+unsafe impl Send for NativeGLContextHandle {}
+
+impl NativeGLContextMethods for NativeGLContext {
+    type Handle = NativeGLContextHandle;
+
+    fn get_proc_address(addr: &str) -> *const () {
+        let addr = CString::new(addr.as_bytes()).unwrap().as_ptr();
+        unsafe {
+            let p = wgl::GetProcAddress(addr) as *const _;
+            if !p.is_null() { return p; }
+            match *GL_LIB {
+                Some(ref lib) =>  kernel32::GetProcAddress(lib.0, addr) as *const _,
```

Comment:
It seems strange but we have to load and mix symbols from both because wglGetProcAddress​ doesn't return function pointers from some legacy functions that are exported by the opengl32.dll itself (old methods coming from OpenGL 1.1). I'll add some comments to the code to explain this.